File: weekly_contest/Weekly 514/q1.py
class Solution:
    def minPrice(self, prices: list[int], discounts: list[int]) -> float:

        # Sort prices and discounts in ascending order
        prices.sort(reverse=True)
        discounts.sort(reverse=True)

        # Iterate prices and discounts
        ans = 0
        curr = 0
        for p, d in zip(prices, discounts):
            ans += (p * (100 - d)) / 100
            curr += 1

        ans += sum(prices[curr:])
        return ans


if __name__ == "__main__":

    solution = Solution()

    # Test cases 1
    print(solution.minPrice([10, 30, 21], [50, 60])) # Output: 32.50000

    # Test cases 2
    print(solution.minPrice(prices = [100,70], discounts = [10,40,50])) # Output: 92.00000

    # Test cases 3
    print(solution.minPrice(prices = [7,3,9], discounts = [100,100])) # Output: 3.00000

File: WeeklyContest/Weekly 514/q1.java
import java.util.Arrays;

class Solution {
    public double minPrice(int[] prices, int[] discounts) {

        // Sort prices and discounts first
        Arrays.sort(prices);
        Arrays.sort(discounts);

        // Get length of prices and discounts
        int pricesLength = prices.length;
        int discountsLength = discounts.length;

        // Count by price and discount and add upp
        double ans = 0;
        for (int i = pricesLength - 1; i >= 0; i--) {
            if (discountsLength > 0) {
                int p = prices[i];
                int d = discounts[--discountsLength];
                ans += (double) (p * (100 - d)) / 100;
            } else {
                ans += prices[i];
            }
        }

        return ans;
    }

    public static void main(String[] args) {
        
        Solution solution = new Solution();

        // Test case 1
        int[] prices = {10, 30, 21};
        int[] discounts = {50, 60};
        System.out.println(solution.minPrice(prices, discounts));
    }
}